### ISSUE TITLE
Keyboard German to German Raspberry PI4 not working

### ISSUE THREAD

**Issue by cw-felix** [rustdesk/rustdesk#9991]
### Bug Description

Hello,

If i connect with Rustdesk to a Raspberry PI4 local Language German: 
![image](https://github.com/user-attachments/assets/829a2e66-7f81-4932-a27d-4a0138f4e0d2)
The Keys are not forwarded correctly. For example instead of * i get }.  Do you have a Idea how to fix this?

Thanks

### How to Reproduce

Install Rustdesk under Raspbian and connect.

### Expected Behavior

Correct keys.

### Operating system(s) on local (controlling) side and remote (controlled) side

Windows 10 -> Raspbian

### RustDesk Version(s) on local (controlling) side and remote (controlled) side

1.3.1 - > 1.2.7

### Screenshots

![image](https://github.com/user-attachments/assets/6cafb1af-a945-4be2-b9b8-258ed8f57995)

**Comment by rustdesk** [comment]
Please try out 1.3.2, it might be fix there.

**Comment by cw-felix** [comment]
Same issue with Version 1.3.2 on both Systems. 

Translate mode is working but we allways have to change the keyboard input method.

**Comment by rustdesk** [comment]
Have you tested on the other devices? only RPI have this issue? We have no RPI devices, it is not easy for us to reproduce it.

**Comment by rustdesk** [comment]
> we allways have to change the keyboard input method.

What does this mean? Can not you stick to translate mode?

**Comment by cw-felix** [comment]
> Have you tested on the other controlled devices? only RPI have this issue? We have no RPI devices, it is not easy for us to reproduce it.

Yes other devices work fine. I can provide a test system for you if you want so.

**Comment by cw-felix** [comment]
> > we allways have to change the keyboard input method.
> 
> What does this mean? Can not you stick to translate mode?

It will only stick for the device i changed it for. I did not find a global setting. We have about 300 devices and 15 users so we would have to change the input method about 4500 times.

**Comment by rustdesk** [comment]
@fufesou figure it out how to change this via https://rustdesk.com/docs/en/self-host/client-configuration/advanced-settings/

**Comment by rustdesk** [comment]
@fufesou also find a way to test RPI4. RPI4 is more and more popular, we can not ignore it.

**Comment by fufesou** [comment]
> We have about 300 devices and 15 users so we would have to change the input method about 4500 times.

We will fix `RPI4`.

But for now, you can change the keyboard mode by running a command.

1. Find the config's directory.
```
Windows: %appdata%\rustdesk\config\peers
Linux: ~/.config/rustdesk/peers
MacOS: ~/Library/Preferences/com.carriez.RustDesk/peers
```
2. Execute the command to change `keyboard_mode` of all peers to `translate`. We use two options, `map` and `translate` as the `keyboard_mode`.

Windows
```
Get-ChildItem -Path "PATH-OF-THE-CONFIG" -Recurse -Filter *.toml | ForEach-Object {
    (Get-Content $_.FullName) -replace "keyboard_mode = '.*'", "keyboard_mode = 'translate'" | Set-Content $_.FullName
}
```

Linux & MacOS
```bash
  find PATH-OF-THE-CONFIG -type f -name "*.toml" -exec sed -i "s/keyboard_mode = '.*'/keyboard_mode = 'translate'/g" {} +
```

**Comment by slumbergod** [comment]
I'm encountering a similar issue (I can start a new issue if someone complains it is unrelated)

I am using a rustdesk client to client connection over tailscale. Both of the machines are running Manjaro Linux in English but both have Spanish keyboards. The rustdesk clients are both 1.3.5

When I type a command in the terminal of the remote machine rustdesk symbols like "-" are wrong. Rustdesk seems to be sending English keyboard symbols. I've tried the three modes but none of them make a difference (legacy, map, and translate).

Is there a way to correct this? I'm new to rustdesk but I can't find an obvious solution. Thanks.

**Comment by cw-felix** [comment]
@fufesou Hi, is there any new information on this? This issue is still a big problem for us.

**Comment by fufesou** [comment]
Sorry, I failed to install a PI VM.

**Comment by cw-felix** [comment]
@fufesou I can provide a Raspberry PI with remote access. Does this help you?

**Comment by fufesou** [comment]
@cw-felix Thanks. Please send me an email `bGlubG9uZzEyNjZAZ21haWwuY29t`

**Comment by issteve** [comment]
I have the same problem and couldn't find a way - of all the keyboard input mode options -   on rustdesk 1.3.9 to make a German keyboard from a Xubuntu or Ubuntu Mate PC trying to send the keys to a Xubuntu or Ubuntu Mate PC work correctly (and not acting as an English keyboard). 
Both on 24.04.2 LTS (most likely still on X11 not wayland) and rustdesk 1.3.9. Sending the keys from the Xubuntu-Laptop to Windows 11 (both 1.3.9, too) works correctly.

https://github.com/rustdesk/rustdesk-server/issues/488 looks similar.

**Comment by slumbergod** [comment]
It looks like my contributions to this problem have been deleted (thanks admins!) so I will add my 2 cents worth again. 
- This issue of keyboard mapping affects me using manjaro too (latest updates applied and using latest rustdesk). I use English but my keyboards on my machines are all Mexican Spanish.
- Rustdesk starts when the system starts. If I connect to a remote machine the mapping is always wrong initially (the remote machine seems to receive the English key rather than the actual one from the Spanish keyboard). It makes no difference what keyboard mapping option is chosen.

But in the last week I noticed something. If I close the remote window and then reopen it, the mapping works correctly. I use rustdesk daily and I have tested this every day successfully. No idea why but maybe that helps someone else.

**Comment by slumbergod** [comment]
Text is now mapping correctly for me after this morning's update to RustDesk 1.4.0
Thank you!

**Comment by fufesou** [comment]
For those who need confirmation.
You can run `xmodmap -pke | grep 35` on the terminal.
The screenshot shows the layout is `en`. Then after running `setxkbmap de` , the layout is `de`.

![Image](https://github.com/user-attachments/assets/898d0e03-6be2-481e-b595-4adddd7d5b7e)
